app/Filament/Resources/ServerResource/Pages/EditServer.php

```diff
@@ -460,6 +492,66 @@ public function form(Form $form): Form
                                     ])
                                     ->required(),
 
+                                Forms\Components\TagsInput::make('ports')
+                                    ->columnSpan(3)
+                                    ->placeholder('Example: 25565, 8080, 1337-1340')
+                                    ->splitKeys(['Tab', ' ', ','])
+                                    ->helperText(new HtmlString('
+                                        These are the ports that users can connect to this Server through.
+                                        <br />
+                                        You would typically port forward these on your home network.
+                                    '))
+                                    ->label('Ports')
+                                    ->formatStateUsing(fn (Server $server) => $server->ports->map(fn ($port) => (string) $port)->all())
+                                    ->afterStateUpdated(self::ports(...))
+                                    ->live(),
+
+                                Forms\Components\Repeater::make('portVariables')
```

Comment:
If you don't select an optional port it will auto select the first port:
![grafik](https://github.com/user-attachments/assets/714137bc-a658-4aa2-9572-0fad1c5f2555)

This should not happen.